src/Feature/SignatureHelp/Feature_SignatureHelp.re

```diff
@@ -348,6 +361,41 @@ let update = (~maybeBuffer, ~maybeEditor, ~extHostClient, model, msg) =>
       },
       Nothing,
     )
+  | CursorMoved(editorID) =>
```

Comment:
Since we have to keep an eye on different events to decide when to show / retrigger signature help, that suggests maybe a subscription could streamline and simplify things (by automatically reacting to state changes). 

We don't have a good example to point to for the language services, but something to think about..

Reply:
Thats an interesting idea -- I'm very unfamiliar with Subscriptions in general, but nothing I can't learn! Let me know if you want me to redo this PR to use one.